firewood/src/shale/compact.rs

```diff
@@ -44,13 +46,19 @@ impl Storable for CompactHeader {
                 size: Self::MSIZE,
             })?;
         #[allow(clippy::indexing_slicing)]
-        let payload_size =
-            u64::from_le_bytes(raw.as_deref()[..8].try_into().expect("invalid slice"));
+        let payload_size = u64::from_le_bytes(
```

Comment:
We're doing all this work to specify a specific endianness and I think that's not necessary. I think we should be using the native endianness and just write the bytes out. This becomes something like:
```rust
        let payload_size = *bytemuck::cast_slice(&derefed[..Self::IS_FREED_OFFSET]).first().expect("wrong size");
```

You can probably do this with `bytemuck::cast` which would be more readable.